### PR TITLE
Support setting filter expirations

### PR DESCRIPTION
**Pull request by Tak**
- Shows the remaining filter duration in the filter list view
- Allows setting/updating durations for filters

### CODE REVIEW

File: app/src/main/java/com/keylesspalace/tusky/network/MastodonApi.kt [hunk 1]
```diff
@@ -536,8 +536,8 @@ interface MastodonApi {
         @Field("context[]") context: List<String>,
         @Field("irreversible") irreversible: Boolean?,
         @Field("whole_word") wholeWord: Boolean?,
-        @Field("expires_in") expiresIn: Int?
-    ): Call<Filter>
+        @Field("expires_in") expiresInSeconds: Int?
+    ): NetworkResult<Filter>
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/network/MastodonApi.kt:540]: all the methods returning `NetworkResult<*>` must be `suspend fun` or they will fail with `NetworkOnMainThreadException`

File: app/src/main/java/com/keylesspalace/tusky/FiltersActivity.kt [hunk 2]
```diff
@@ -47,119 +47,100 @@ class FiltersActivity : BaseActivity() {
...
+            filters.map { filter ->
+                if (filter.expiresAt == null) {
+                    filter.phrase
+                } else {
+                    getString(
+                        R.string.filter_expiration_format,
+                        filter.phrase,
+                        DateUtils.getRelativeTimeSpanString(
+                            filter.expiresAt.time,
+                            Date().time,
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/FiltersActivity.kt:135]:
```suggestion
                            System.currentTimeMillis(),
```

(saves one object allocation)

File: app/src/main/java/com/keylesspalace/tusky/view/FilterDialog.kt [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        .setNegativeButton(R.string.filter_dialog_remove_button) { _, _ ->
+            activity.deleteFilter(itemIndex)
+        }
+        .setNeutralButton(android.R.string.cancel, null)
+        .show()
+}
+
+// Mastodon *stores* the absolute date in the filter,
+// but create/edit take a number of seconds (relative to the time the operation is posted)
+fun getSecondsForDurationIndex(index: Int, context: Context?, default: Date? = null): Int? {
```
Comment (connyduck) [inline app/src/main/java/com/keylesspalace/tusky/view/FilterDialog.kt:67]: I think context can be non-null

**Review by connyduck — COMMENTED** [review]
Can the edit dialog be a little less dense? A bit of padding between elements and to the frame, the spinner on its own line, `lineSpacingMultiplier="1.1"` on the TextView?

File: app/src/main/java/com/keylesspalace/tusky/view/FilterDialog.kt (hunk 3 above)
Comment (Tak) [reply app/src/main/java/com/keylesspalace/tusky/view/FilterDialog.kt:67]: It's getting called with null in the tests

**Comment by Tak** [comment]
Updated

**Review by connyduck — APPROVED** [review]

**Pull request merged**